pkg/validator/am0014/verify_secrets.go

```diff
@@ -0,0 +1,102 @@
+package am0014
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/types"
+	"github.com/mt-sre/addon-metadata-operator/pkg/validator"
+	"gopkg.in/yaml.v3"
+)
+
+func init() {
+	validator.Register(NewVerifySecretParams)
+}
+
+const (
+	code = 14
+	name = "verify_secret_params"
+	desc = "Ensure that `secretParameters` present in app-interface's SaaS file match secrets in the addon metadata"
+)
+
+type Response struct {
+	SecretParameters []SecretParameter `json:"secretParameters,omitempty" yaml:"secretParameters,omitempty"`
+}
+
+type SecretParameter struct {
+	Name   string `json:"name,omitempty" yaml:"name,omitempty"`
+	Secret struct {
+		Path  string `json:"path,omitempty" yaml:"path,omitempty"`
+		Field string `json:"field,omitempty" yaml:"field,omitempty"`
+	}
+}
+
+func NewVerifySecretParams(deps validator.Dependencies) (validator.Validator, error) {
+	base, err := validator.NewBase(
+		code,
+		validator.BaseName(name),
+		validator.BaseDesc(desc),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &VerifySecretParams{
+		Base: base,
+	}, nil
+}
+
+type VerifySecretParams struct {
+	*validator.Base
+}
+
+func (i *VerifySecretParams) Run(ctx context.Context, mb types.MetaBundle) validator.Result {
+
```

Comment:
Gotta skip the check (return Success) when mb.AddonMeta.Secrets is nil because it's an optional attribute